fcs/build.fsx

```diff
@@ -77,11 +77,15 @@ Target "Restore" (fun _ ->
     // We assume a paket restore has already been run
     runDotnet __SOURCE_DIRECTORY__ "restore FSharp.Compiler.Service.sln -v n"
     for p in [ "../packages.config" ] do
-        ExecProcess (fun info ->
-            info.FileName <- FullName @"./../.nuget/NuGet.exe"
-            info.WorkingDirectory <- FullName @"./.."
-            info.Arguments <- sprintf "restore %s -PackagesDirectory \"%s\" -ConfigFile \"%s\""   (FullName p) (FullName "./../packages") (FullName "./../NuGet.Config")) TimeSpan.MaxValue
-        |> assertExitCodeZero           
+        let rec executeProcess count =
+            let result = ExecProcess (fun info ->
+                info.FileName <- FullName @"./../.nuget/NuGet.exe"
+                info.WorkingDirectory <- FullName @"./.."
+                info.Arguments <- sprintf "restore %s -PackagesDirectory \"%s\" -ConfigFile \"%s\""   (FullName p) (FullName "./../packages") (FullName "./../NuGet.Config")) TimeSpan.MaxValue
+            if result <> 0 && count > 1 then  executeProcess (count - 1) else 0
```

Comment:
This should read `... else result` so that if the 5 tries are exhausted the last error code is returned.